### ISSUE TITLE
setIndented(false) also removes new lines from text in views

### ISSUE THREAD

**Issue by shimikano** [xmlet/HtmlFlow#104]
Similarly to #46, I tried to disable indentation in conjunction with a `<textarea>` element to avoid unwanted whitespace.

However, I noticed that `setIndented(false)` for views also removes new lines in `text()` blocks. This is problematic, e.g., for `<script>` tags. The following test fails, with the rendered view containing no new lines at all:
```java
  @Test
  public void viewWithNoIndent() {
    var view = HtmlFlow.view(page -> {
      page
        .div()
          .textarea()
            .text("Sample text\nfoo\nbar")
          .__()
          .script()
            .text("""
              // some comment
              console.log('Hello world');""")
          .__()
        .__();
    }).setIndented(false);

    var actual = view.render();

    var expected = """
      <div><textarea>Sample text
      foo
      bar</textarea><script>// some comment
      console.log('Hello world');</script></div>""";

    assertEquals(expected, actual);
  }
```

For docs, this seems not to be the case (the test passes):
```java
  @Test
  public void docWithNoIndent() {
    var stringBuilder = new StringBuilder();
    HtmlFlow.doc(stringBuilder).setIndented(false)
      .div()
        .textarea()
          .text("Sample text\nfoo\nbar")
        .__()
        .script()
          .text("""
            // some comment
            console.log('Hello world');""")
          .__()
      .__();

    var actual = stringBuilder.toString();

    var expected = """
      <div><textarea>Sample text
      foo
      bar</textarea><script>// some comment
      console.log('Hello world');</script></div>""";

    assertEquals(expected, actual);
  }
```

Is the view behavior a bug or is there another way to preserve the new lines in `text()` blocks?

Thank you very much.

**Comment by fmcarvalho** [comment]
Well, it seems the way we are providing the `setIndentation` for `HtmView` is not correct. The way `HtmlDoc` and `HtmView` work are completely different. Indentation in `HtmlDoc` is made though the oldest `newlineAndIndent()`: 
https://github.com/xmlet/HtmlFlow/blob/4aac871bf91e26d75532152a40d375b09e708c69/src/main/java/htmlflow/visitor/HtmlVisitor.java#L122 

Whereas `HtmView` is always built with indentation set to `true`, and later on render we decide wether we want to remove it. The bug is on the following lines from `HtmlContinuationSyncStatic` class:
```java
String block = v.isIndented
    ? staticHtmlBlock
    : stream(staticHtmlBlock.split(lineSeparator())).map(String::trim).collect(joining());
```

We are wrongly removing all line separators. To fix that we have to remove this code. Also we have to change the way we were building an `HtmlView`. Basically, whenever we call `setIndent(...)`we have to run the `preprocessing(...)` from the scratch with the correct indentation flag.  And to do it we need to keep the `HtmTemplate` function associated with the `HtmlView`. 

Is not very complicated and I think it has no impact on performance because all that will be made to run on preprocessing. Moreover I never liked that if/else (i.e ternary) on `HtmlContinuationSyncStatic`.

However this refactoring includes a couple of steps and I need to review all this again. I hope to dedicate some time to it at the end of this week.

Thank you,
Miguel

**Comment by shimikano** [comment]
Thank you, Miguel, for your efforts 🙏

**Comment by fmcarvalho** [comment]
Solved on new release 4.2

**Comment by fmcarvalho** [comment]
@shimikano As you may be aware, I am the author and primary advocate for the HtmlFlow Java library. As part of my continuous endeavors, I am required to provide justification for this project to my university.

HtmlFlow serves as a versatile auxiliary tool utilized across various sectors, facilitating software development by streamlining tasks such as HTML reporting automation, web templating, and more. 

If you have utilized HtmlFlow, I kindly request your testimony regarding my dedication to addressing issues, implementing new features, or otherwise aligning with your requirements and/or your company within the scope of a specific project, task, or work that utilized HtmlFlow during a certain timeframe.

Your testimony will play a crucial role in demonstrating my work to my university.

Please feel free to send your testimonial to my email address at ISEL: [miguel.gamboa@isel.pt](mailto:miguel.gamboa@isel.pt)

Thank you in advance for your attention and assistance.
